src/gc/gcinterface.h

```diff
@@ -451,6 +464,23 @@ class IGCHeap {
     // Walks an object, invoking a callback on each member.
     virtual void WalkObject(Object* obj, walk_fn fn, void* context) = 0;
 
+    // Walk the heap object by object.
+    virtual void WalkHeap(walk_fn fn, void* context, int gen_number, BOOL walk_large_object_heap_p) = 0;
+    
+    // Walks the survivors and get the relocation information if objects have moved.
+    virtual void WalkSurvivors(void* gc_context, record_surv_fn fn, size_t diag_context, walk_surv_type type) = 0;
```

Comment:
this is because it's that way in the profiling/ETW code....they use void* sometimes and size_t other times (different kinds of contexts) and I didn't change that.

---
In reply to: [89092893](https://github.com/dotnet/coreclr/pull/8199#discussion_r89092893) [](ancestors = 89092893)